### ISSUE TITLE
data_aspect and responsive do not work together when data_aspect>1

### ISSUE THREAD

**Issue by basnijholt** [holoviz/holoviews#5360]
#### ALL software version info
- HoloViews tried latest master and latest release
- Bokeh 2.4.2

#### Description of expected behavior and the observed behavior
data_aspect and responsive do not work together when data_aspect>1.

The width of the plot seems to get set to 0.

#### Complete, minimal, self-contained example code that reproduces the issue

Here I am plotting the figure twice, once with just HoloViews and once by first rendering and then calling `bokeh.io.show`

```python
import holoviews as hv
import bokeh.io

hv.notebook_extension("bokeh")


i = 1 + 1e-3  # Change to just 1, and the problem is gone!

p = hv.Polygons({"x": [0, 1, 1, 0], "y": [0, 0, i, i]}).opts(
    responsive=True, data_aspect=1, min_height=50
)
display(p)  # shows messed-up figure
bokeh.io.show(hv.render(p, "bokeh"))  # correctly shows figure
```

#### Screenshots or screencasts of the bug in action
<img width="877" alt="image" src="https://user-images.githubusercontent.com/6897215/180275220-415a2030-831e-4b6f-a5ee-882ce40a7003.png">

**Comment by Hoxbro** [comment]
**TL:DR:** This is properly a bokeh problem and has been fixed upstream. 

---

So I can somewhat recreate this problem. I'm using `bokeh` 2.4.3 and the latest master of `holoviews`. The difference is either that they both fail or both succeed.  

With `i = 1.001`
![image](https://user-images.githubusercontent.com/19758978/181738013-e026d41c-2b09-4185-a1fa-04764bf47216.png)


HoloViews uses a different (older?) version of defining a plot's dimension and aspect properties. These properties are then converted to Bokeh definition with this function:

https://github.com/holoviz/holoviews/blob/67f1fcf35586fdd2e9d83786e3110f87310123dd/holoviews/plotting/bokeh/util.py#L185-L188

So comparing Bokeh properties for  `i = 1` and `i = 1.001`,

![image](https://user-images.githubusercontent.com/19758978/181738561-4bea1572-a49d-461b-a62b-2a042f6fc4a2.png)

![image](https://user-images.githubusercontent.com/19758978/181738618-ced88373-4331-4aec-bbd0-6bd71124d541.png)

Where the relevant properties are: `min_height`, `sizing_mode`, and `aspect_ratio`. This can then be converted to bokeh only example:
``` python
from bokeh.plotting import figure, output_file, show
import bokeh.io

# instantiating the figure object
graph = figure(min_height=50, sizing_mode="scale_both", aspect_ratio=0.999)
    
# the points to be plotted
xs = [[[[0, 1, 1, 0]]]]
ys = [[[[0, 0, 1, 1]]]]
     
# plotting the graph
graph.multi_polygons(xs, ys)

# displaying the model
bokeh.io.show(graph)
```

This will fail for some combinations and not for other combinations: 

| Failure | Success |
| ------- | --------- |
| ![image](https://user-images.githubusercontent.com/19758978/181739274-a4e04006-2e94-48df-a3a6-3164fca2c521.png) | ![image](https://user-images.githubusercontent.com/19758978/181739313-52f92531-9c73-42c4-b848-c24b5b8814ba.png) |

Luckily, this has been fixed by Bokeh. Running the example which failed on the latest dev release, we get:
![Screenshot 2022-07-29 at 11-10-00 Bokeh Plot](https://user-images.githubusercontent.com/19758978/181739659-b90b8df0-209d-44eb-ab8e-3a264b4c3d3c.png)

**Comment by basnijholt** [comment]
Thanks a lot for your detailed answer @Hoxbro! Are you aware of which Bokeh issue/PR solved this problem?

**Comment by Hoxbro** [comment]
No, the new version of Bokeh has an entirely new layout engine, which is properly what fixed it. 

It could be a single PR that has fixed it, but I haven't looked.

**Comment by basnijholt** [comment]
@Hoxbro, is it already possible to use the new bokeh with HoloViews?

I have tried but there seems to be an incompatibility with panel.

```python
File ~/Work/funcs.py:1893, in plot(device, min_aspect, padding, frame_width, responsive, merged_layers, custom_color, hover)
   1883 def plot(
   1884     device,
   1885     min_aspect: float = 0.25,
   (...)
   1891     hover: bool = True,
   1892 ):
-> 1893     import holoviews as hv
   1894     from bokeh.models import HoverTool
   1896     hv.extension("bokeh", logo=False)

File ~/mambaforge/lib/python3.9/site-packages/holoviews/__init__.py:10, in <module>
      6 __version__ = str(param.version.Version(fpath=__file__, archive_commit="$Format:%h$",
      7                                         reponame="holoviews"))
      9 from . import util                                       # noqa (API import)
---> 10 from .annotators import annotate                         # noqa (API import)
     11 from .core import archive, config                        # noqa (API import)
     12 from .core.boundingregion import BoundingBox             # noqa (API import)

File ~/mambaforge/lib/python3.9/site-packages/holoviews/annotators.py:8, in <module>
      4 from inspect import getmro
      6 import param
----> 8 from panel.pane import PaneBase
      9 from panel.layout import Row, Tabs
     10 from panel.util import param_name

File ~/mambaforge/lib/python3.9/site-packages/panel/__init__.py:67, in <module>
     65 from .pane import panel, Pane # noqa
     66 from .param import Param # noqa
---> 67 from .template import Template # noqa
     68 from .widgets import indicators # noqa
     70 __all__ = (
     71     "__version__",
     72     "Accordion",
   (...)
    102     "widgets",
    103 )

File ~/mambaforge/lib/python3.9/site-packages/panel/template/__init__.py:2, in <module>
      1 from ..config import _config
----> 2 from .base import Template, BaseTemplate # noqa
      3 from .bootstrap import BootstrapTemplate # noqa
      4 from .fast import FastListTemplate, FastGridTemplate # noqa

File ~/mambaforge/lib/python3.9/site-packages/panel/template/base.py:42, in <module>
     40 from ..widgets import Button
     41 from ..widgets.indicators import BooleanIndicator, LoadingSpinner
---> 42 from .theme import THEMES, DefaultTheme, Theme
     44 if TYPE_CHECKING:
     45     from bokeh.document import Document

File ~/mambaforge/lib/python3.9/site-packages/panel/template/theme/__init__.py:56, in <module>
     51     __abstract = True
     54 BOKEH_DARK = dict(_dark_minimal.json)
---> 56 BOKEH_DARK['attrs']['Figure'].update({
     57     "background_fill_color": "#3f3f3f",
     58     "border_fill_color": "#2f2f2f",
     59 })
     61 class DarkTheme(Theme):
     62     """
     63     The DefaultTheme uses the standard Panel color palette.
     64     """

KeyError: 'Figure'
```

**Comment by Hoxbro** [comment]
Running the current main of Panel and HoloViews with bokeh3.1.dev2, we get the following output

So soon you will be able to run this example without it collapsing  :slightly_smiling_face: 

![image](https://user-images.githubusercontent.com/19758978/217811082-25780aae-3d1a-4596-8231-e3563eb26e5e.png)

**Comment by basnijholt** [comment]
That's great! I am looking forward to Bokeh 3 support 😄
